gst/nnstreamer/ml_agent.c

```diff
@@ -78,29 +82,74 @@ mlagent_parse_uri_string (const GValue * val)
       /** Convert the given URI for a model to the file path */
       g_autofree gchar *name = g_strdup (parts[MODEL_PART_IDX_NAME]);
       guint version = strtoul (parts[MODEL_PART_IDX_VERSION], NULL, 10);
-      g_auto (GStrv) stringfied_json = NULL;
-      GError *err;
+      g_autofree gchar *stringfied_json = NULL;
+      g_autoptr (JsonParser) json_parser = NULL;
       gint rcode;
 
       /**
        * @todo The specification of the data layout filled in the third
        *       argument (i.e., stringfied_json) by the callee is not fully decided.
        */
-      rcode = ml_agent_model_get (name, version, stringfied_json, &err);
+      rcode = ml_agent_model_get (name, version, &stringfied_json, &err);
+      if (rcode != 0) {
+        nns_loge
+            ("Failed to get the stringied JSON using the given URI(%s): %s",
+            uri, (err ? err->message : "unknown reason"));
+        goto fallback;
```

Comment:
Are you going to return the URI string even when there is an error?

Reply:
Yes. If the given string is not for ml-agent, ```_gtfc_setprop_MODEL``` is in charge of it when it did before.

Reply:
Then, how the caller will be notified for such errors? (you need to specify it in the doxygen, too)

Reply:
This function converts a given URI to a model path c-string when ml-agent is enabled and the given URI is valid. When the ml-agent is disabled, it works in the same way as ```g_value_dup_string``` works (please see ```ml_agent.h```). As ```g_value_dup_string``` does, it returns ```NULL``` when the ```GValue *val``` does not hold a string value. If the string has been verified as a valid URI, then this function converts it to a path c-string. Otherwise, it just returns the string that the ```*val``` holds.